### ISSUE TITLE
Protocol buffer import capability issues

### ISSUE THREAD

**Issue by ericzhang6222** [arr-ai/arrai#496]
#478 is done, and there are 3 pending issues still:
https://github.com/arr-ai/arrai/issues/478#issuecomment-654186357
https://github.com/arr-ai/arrai/issues/478#issuecomment-654567482
https://github.com/arr-ai/arrai/issues/478#issuecomment-657423286

**Comment by orlade-anz** [comment]
Once this is decided it will likely break existing protobuf importing arr.ai code, so we should sort it out sooner rather than later.

**Comment by marcelocantos** [comment]
The signature of the protobuf API should be:

```arrai
//encoding.proto.decode(descriptor <: tuple, messageTypeName <: string, messageBytes <: bytes) <: tuple
//encoding.proto.encode(descriptor <: tuple, messageTypeName <: string, message <: tuple) <: bytes
```

If `descriptor` is `//encoding.proto.proto`, the output of `//encoding.proto.decode` should be usable as the first parameter in calls to `//encoding.proto.decode` and `//encoding.proto.encode`.

**Comment by orlade-anz** [comment]
So to clarify, when decoding `sysl.pb` in order to get a file descriptor with which to decode encoded sysl protos, you would use:
```arrai
let sysl = //encoding.proto.decode(//encoding.proto.proto, "FileDescriptorSet", //os.file("sysl.pb"));
let model = //encoding.proto.decode(sysl, "Module", //os.file("model.pb"));
model
```

**Comment by marcelocantos** [comment]
Yes, and to help with bootstrapping, a convenience wrapper might be in order:
```arrai
//encoding.proto.descriptor(//os.file("sysl.pb"))
```
which would be equivalent to
```arrai
//encoding.proto.decode(//encoding.proto.proto, "FileDescriptorSet", //os.file("sysl.pb"))
```

**Comment by ericzhang6222** [comment]
As discussed, remove `//encoding.proto.proto`, and the code looks like `//encoding.proto.decode(//encoding.proto.descriptor(//os.file("sysl.pb")), "Module", //os.file("model.pb"))`

**Comment by ericzhang6222** [comment]
`//encoding.proto.descriptor(//os.file("sysl.pb"))` implementation should be changed still to build and return a real tuple.

**Comment by marcelocantos** [comment]
1. https://github.com/arr-ai/arrai/issues/478#issuecomment-654567482: See #550.
1. https://github.com/arr-ai/arrai/issues/478#issuecomment-654186357 is probably unresolvable. Let's leave it for now.
1. https://github.com/arr-ai/arrai/issues/478#issuecomment-654186357: This is pending, afaik. I want to be able to build a proto descriptor structure in arr.ai programmatically and use it to import a .proto file. This isn't a high priority, so we'll keep this issue open at P1.
